### DISCUSSION TITLE
Building IOS app help

### DISCUSSION THREAD

**Discussion by Daanvdplas** [fzyzcjy/flutter_rust_bridge#1063] [Q&A]
Hello everyone,

I'm trying to make an IOS app that uses wasmi. Wasmi allows WebAssembly modules to be loaded in binary format and their functions invoked. When I try to `cargo lipo` it gives me this error I think there is a correlation:
![Screenshot 2023-02-15 at 09 55 28](https://user-images.githubusercontent.com/93204684/218980233-e4211630-852a-4d4b-9790-bdbd5cbb440d.png)

Honestly, I'm very new to flutter and this repo and I couldn't find anything similar so this error. Any ideas what could be wrong?

**Comment by Daanvdplas** [comment]
Is there perhaps an alternative to cargo lipo?

**Comment by fzyzcjy** [comment]
I am not familiar with wasmi. Looks like you want Rust code to be compiled into WASM? Then maybe look at how flutter_rust_bridge compiles into the Flutter Web - where it is compiled to wasm

**Reply by Daanvdplas** [reply]
Other way around, I have a WASM blob and wasmi is the interpreter to execute it natively. 

I got it working on android so I think lipo can't handle it for some reason

**Reply by fzyzcjy** [reply]
I see. Then I do not have experience about that. Given flutter_rust_bridge is nothing but a code generator (and a thin runtime), maybe search how to do it for *general* Rust/Flutter?

**Comment by Daanvdplas** [comment]
I'm now trying to just build on an aarch64 platform. After following all the steps from the flutter_rust_bridge everything is working until I do flutter run and I get this error:

`Xcode build done.`
`Failed to build iOS app`
`Could not build the precompiled application for the device`.
`Swift Compiler Error (Xcode): Cannot find 'GeneratedPluginRegistrant' in scope`
`/Users/parity/Documents/trappist-extra/ios/Runner/AppDelegate.swift:11:4`

I already tried `flutter clean`.

Help is much appreciated :)

**Reply by fzyzcjy** [reply]
That looks like a pure-Flutter problem, GeneratedPluginRegistrant is unrelated to frb.

Maybe `flutter pub get`

**Reply by Daanvdplas** [reply]
I'm reading a lot about pod (Cacaupods) but I can't find any related files in my ios folder. Should there be?

**Reply by fzyzcjy** [reply]
There are a few, but not sure which file you refer to. Maybe create a brand new flutter project to see files structures

**Reply by Daanvdplas** [reply]
I got it running I got the settings for macos and ios mixed up.

If I want to run it on macos now, do I open `ios/Runner.xcodeproj` or `macos/Runner.xcodeproj` in this [step](https://cjycode.com/flutter_rust_bridge/integrate/ios_linking.html)?

**Reply by Daanvdplas** [reply]
As well as [here](https://cjycode.com/flutter_rust_bridge/tutorial_with_flutter.html#ios-app), do I need to copy the static library to `ios/Runner/` or `macos/Runner/` or just leave it where it is?

**Reply by fzyzcjy** [reply]
cc @Desdaemon who create that helpful tutorial :)

 I guess you can also search a bit more about general flutter to rust integrations

**Reply by Desdaemon** [reply]
Also how are you bundling your WASM blob with the app? If you can have it contained within `native` it should still be able to use the steps I described for static linking.

**Reply by fzyzcjy** [reply]
Btw not sure whether related: "native assets" https://github.com/dart-lang/sdk/issues/50565

**Reply by Daanvdplas** [reply]
First, thanks a lot for your help, without you guys this would be a lot more difficult!

So I got it running on my Iphone, and from @Desdaemon's comment I understand that it should be able to run on my mac as well with the static linking. However, if I do flutter run I get:
>══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY 
The following ArgumentError was thrown building Builder:
Invalid argument(s): Failed to lookup symbol 'store_dart_post_cobject': dlsym(RTLD_DEFAULT,
store_dart_post_cobject): symbol not found
The relevant error-causing widget was:
  MaterialApp MaterialApp:file:///Users/parity/Documents/trappist-extra/lib/main.dart:18:12

I tried [stripping](https://github.com/fzyzcjy/flutter_rust_bridge/issues/250), I added the `bridge_generated.h` mentioned [here](https://cjycode.com/flutter_rust_bridge/integrate/ios_headers.html#macos) and I'm now trying to bundle it as described in the second link all the way at the end for MACOS (first time I am doing something in the macos folder, it is really hard to follow the steps and understand when I need to make changes in macos/Runner or ios/Runner) but now I get this: 
>Undefined symbols for architecture arm64:

It seems I'm not linking correctly, any idea?

**Reply by Desdaemon** [reply]
The symbols might still be stripped depending on how you're calling `dummy_method_to_enforce_bundling`, this process is still somewhat opaque and would appreciate people more knowledgeable about Xcode linking chiming in. Usually you might have to `print` the result as well.

**Reply by Daanvdplas** [reply]
But coming back to these two questions [1](https://github.com/fzyzcjy/flutter_rust_bridge/discussions/1063#discussioncomment-5061389) & [2](https://github.com/fzyzcjy/flutter_rust_bridge/discussions/1063#discussioncomment-5061483)?

**Reply by Desdaemon** [reply]
1. That `GeneratedPluginRegistrant` is not visible may be due to something you may have inadvertently removed. Try removing the `ios` folder and start from scratch.

2. You can't run the iOS project as a MacOS project, since they have separate config that must be repeated. In other words, the tutorial lists instructions for both of them, but you'll need to follow them for each platform to run on.
